File: client/src/models/character/Characteristic.ts
import {Option} from "../../components/common/InputSelectField";

export enum CharacteristicType {
    Brawn = 'Brawn',
    Agility = 'Agility',
    Intellect = 'Intellect',
    Cunning = 'Cunning',
    Willpower = 'Willpower',
    Presence = 'Presence'
}

export interface Characteristic {
    current: number,
    type: CharacteristicType
}

export const getCharacteristicTypes = (): Option[] => {
    return Object.values(CharacteristicType).map((value) => ({value}))
}

File: server/src/main/java/com/github/genraven/genesys/domain/actor/Characteristic.java
package com.github.genraven.genesys.domain.actor;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;

@Data
@AllArgsConstructor
public class Characteristic {
    protected Characteristic() {}
    private Type type;
    private int current;

    @AllArgsConstructor
    @Getter
    public enum Type {
        BRAWN("Brawn"),
        AGILITY("Agility"),
        INTELLECT("Intellect"),
        CUNNING("Cunning"),
        WILLPOWER("Willpower"),
        PRESENCE("Presence");

        @JsonValue
        private final String label;
    }
}
